src/bitExpert/Disco/Proxy/Configuration/ConfigurationGenerator.php

```diff
@@ -108,60 +109,68 @@ public function generate(ReflectionClass $originalClass, ClassGenerator $classGe
         $aliases = [];
         $methods = $originalClass->getMethods(ReflectionMethod::IS_PUBLIC | ReflectionMethod::IS_PROTECTED);
         foreach ($methods as $method) {
-            if (null !== $reader->getMethodAnnotation($method, BeanPostProcessor::class)) {
-                $postProcessorMethods[] = $method->getName();
-                continue;
-            }
-
-            /* @var \bitExpert\Disco\Annotations\Bean $beanAnnotation */
-            $beanAnnotation = $reader->getMethodAnnotation($method, Bean::class);
-            if (null === $beanAnnotation) {
-                throw new InvalidProxiedClassException(
-                    sprintf(
-                        'Method "%s" on "%s" is missing the @Bean annotation!',
-                        $method->name,
-                        $originalClass->name
-                    )
-                );
-            }
+            $methodReflection = new MethodReflection(
+                $method->class,
+                $method->name
+            );
 
-            foreach ($beanAnnotation->getAliases() as $beanAlias) {
-                $alias = $beanAlias->isTypeAlias() ? (string) $method->getReturnType() : $beanAlias->getName();
-                if (empty($alias)) {
-                    continue;
-                }
+            $beanAnnotation = $reader->getMethodAnnotation($method, BeanPostProcessor::class);
+            if ($beanAnnotation instanceof BeanPostProcessor) {
+                $postProcessorMethods[] = $method->name;
 
-                if (isset($aliases[$alias])) {
+                $proxyMethod = BeanPostProcessorMethod::generateMethod(
+                    $methodReflection,
+                    $beanAnnotation,
+                    $getParameterMethod
+                );
+            } else {
+                /* @var \bitExpert\Disco\Annotations\Bean $beanAnnotation */
+                $beanAnnotation = $reader->getMethodAnnotation($method, Bean::class);
+                if (null === $beanAnnotation) {
                     throw new InvalidProxiedClassException(
                         sprintf(
-                            'Alias "%s" of method "%s" on "%s" is already used by method "%s" of another Bean!'
-                            . ' Did you use a type alias twice?',
-                            $alias,
+                            'Method "%s" on "%s" is missing the @Bean annotation!',
                             $method->name,
-                            $originalClass->name,
-                            $aliases[$alias]
+                            $originalClass->name
                         )
                     );
                 }
 
-                $aliases[$alias] = $method->name;
+                foreach ($beanAnnotation->getAliases() as $beanAlias) {
+                    $alias = $beanAlias->isTypeAlias() ? (string) $method->getReturnType() : $beanAlias->getName();
+                    if (empty($alias)) {
+                        continue;
```

Comment:
I added the check mostly for the return type. Potentially it could not be defined at this stage. Maybe it makes sense to move the BeanMethod generation logic upfront which actually checks for the return type.